portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java

```diff
@@ -1978,6 +1981,27 @@ public void includeLogin(PortalRenderContext rcontext, HttpServletRequest req,
 				rcontext.put("loginUserFirstName", loginUserFirstName);
 				rcontext.put("loginUserDispId", loginUserDispId);
 				rcontext.put("loginUserId", loginUserId);
+
+				boolean displayRole = ServerConfigurationService.getBoolean(PORTAL_DISPLAY_CURRENT_ROLE, true);
+				rcontext.put("currentRole", null);
+				if (displayRole) {
+					String[] parts = getParts(req);
+					String roleId = "";
+					String siteId = (parts.length > 2)? parts[2]: "";
+					if (!StringUtils.startsWith(siteId, "~")) {
+						try {
+							Site site = SiteService.getSite(siteId);
+							roleId = site.getMember(loginUserId).getRole().getId();
+							if (roleId.equals("Student") || roleId.equals("Instructor") || roleId.equals("Teaching Assistant")) {
```

Comment:
Just put this!

rcontext.put("currentRole", rloader.getFormattedMessage("sit_role", roleId));